### ISSUE TITLE
train_expert.py: pad tokens are counted in the loss and never masked (bs>1)

### ISSUE THREAD

**Issue by marksverdhei** [marksverdhei/exex#25]
\`train_expert.py\` tokenizes with \`padding=True\` and sets \`labels = input_ids.clone()\`, and never passes \`attention_mask\`. Pad positions therefore (a) contribute cross-entropy terms (predict \`<pad>\` after the sequence ends) and (b) are attended to by nothing only because padding is right-side — left padding would corrupt real tokens too.

Measured on the tiny fixture with two texts of length 37 and 9: 39 % of the loss terms were pad positions. The s4 26B run used \`--batch_size 2\` on variable-length PubMedQA rows, so its reported task loss and the very noisy loss curve are partly pad noise.

Fix (quick win):
```python
labels = enc.input_ids.clone(); labels[enc.attention_mask == 0] = -100
trainer.train_step(input_ids=enc.input_ids, attention_mask=enc.attention_mask, labels=labels)
```
Also consider masking the first token / BOS and adding a test that asserts pad-invariance of the loss (same texts, different padding → same loss).
